src/api/spec/models/package_spec.rb

```diff
@@ -184,12 +185,18 @@
   end
 
   describe '#service_error' do
-    context 'without error' do
-      it { expect(package_with_service.service_error).to be_nil }
+    def finish_service(package)
+      sleep 0.1 while package.serviceinfo.to_hash['code'] == 'running'
```

Comment:
Why the sleep?

Reply:
I don't know what else do - there is no way to actively wait for the service to finish if it's broken

Reply:
But the service is already broken in the `package_with_broken_service` factory right?

Reply:
yes, but the src server doesn't do the validation

Reply:
if we wanted to check the model function only (you would wonder why as it's nowhere called now) you could mock the backend reply directly.